src/main/scala/com/typesafe/sbt/web/SbtWeb.scala

```diff
@@ -318,22 +304,50 @@ object SbtWeb extends AutoPlugin {
     nodeModules := nodeModuleGenerators(_.join).map(_.flatten).value
   )
 
+  def webJarsPathPrefix: Def.Initialize[Task[String]] = Def.task {
+    path(s"$WEBJARS_PATH_PREFIX/${moduleName.value}/${version.value}/")
+  }
+
+  def syncExportedAssets(track: TrackLevel): Def.Initialize[Task[File]] = Def.task {
+    val syncTargetDir = classDirectory.value
+    val syncRequired = TrackLevel.intersection(track, exportToInternal.value) match {
+      case TrackLevel.TrackAlways =>
+        true
+      case TrackLevel.TrackIfMissing | TrackLevel.NoTracking =>
+        !(syncTargetDir / webJarsPathPrefix.value).exists()
+    }
+    if (syncRequired) {
+      state.value.log.info(s"Exporting ${configuration.value}:${moduleName.value}")
+      syncMappings(streams.value.cacheDirectory, exportedMappings.value, syncTargetDir)
+    } else
+      syncTargetDir
+  }
```

Comment:
Doesn't sbt-web invoke the source generator on [this line](https://github.com/sbt/sbt-web/blob/8d41407586a380efe778f98dff3c5fdae878b329/src/main/scala/com/typesafe/sbt/web/SbtWeb.scala#L266):

```
managedSources := sourceGenerators(_.join).map(_.flatten).value
```

and doesn't `mapping in sources` [here](https://github.com/sbt/sbt-web/blob/8d41407586a380efe778f98dff3c5fdae878b329/src/main/scala/com/typesafe/sbt/web/SbtWeb.scala#L271
) cause dependency to that?:

```scala
sources := managedSources.value ++ unmanagedSources.value,
mappings in sources <<= relativeMappings(sources, sourceDirectories),
```

Alternative Javascript plugins are hooking into `sourceGenerators`, aren't they not?